pkg/pipeline/engine/jenkins/mappper.go

```diff
@@ -39,16 +44,56 @@ func initJenkinsPipelineConverter(execution *v3.PipelineExecution, pipelineSetti
 	if err != nil {
 		return nil, err
 	}
+	memoryRequestSetting, err := pipelineSettingLister.Get(projectID, utils.SettingExecutorMemoryRequest)
+	if err != nil {
+		return nil, err
+	}
+	if err := validateQuantity(memoryRequestSetting.Value); err != nil {
+		return nil, errors.Wrap(err, "invalid executor memory request config")
+	}
+	memoryLimitSetting, err := pipelineSettingLister.Get(projectID, utils.SettingExecutorMemoryLimit)
+	if err != nil {
+		return nil, err
+	}
+	if err := validateQuantity(memoryLimitSetting.Value); err != nil {
+		return nil, errors.Wrap(err, "invalid executor memory limit config")
+	}
+	cpuRequestSetting, err := pipelineSettingLister.Get(projectID, utils.SettingExecutorCPURequest)
+	if err != nil {
+		return nil, err
+	}
+	if err := validateQuantity(cpuRequestSetting.Value); err != nil {
+		return nil, errors.Wrap(err, "invalid executor cpu request config")
+	}
+	cpuLimitSetting, err := pipelineSettingLister.Get(projectID, utils.SettingExecutorCPULimit)
+	if err != nil {
+		return nil, err
+	}
+	if err := validateQuantity(cpuLimitSetting.Value); err != nil {
+		return nil, errors.Wrap(err, "invalid executor cpu limit config")
+	}
 	opts := &executeOptions{
-		gitCaCerts:           cacertSetting.Value,
-		imagePullSecretNames: secretNames,
+		gitCaCerts:            cacertSetting.Value,
+		imagePullSecretNames:  secretNames,
+		executorMemoryRequest: memoryRequestSetting.Value,
```

Comment:
Maybe we need to set a default value for these new settings as the executor won't be able to run if the namespace quota is set.

Reply:
Updated